pkg/collector/collector.go

```diff
@@ -129,26 +165,17 @@ func (collector *Collector) Collect(ch chan<- prometheus.Metric) {
 		ch <- prometheus.MustNewConstMetric(collector.poolsTotal, prometheus.GaugeValue, float(matches[15][1])*1024)
 	}
 
-	version := collector.client.Dial(commands.VERSION)
-	regex = regexp.MustCompile(`((ClamAV)+\s([0-9.]*)/([0-9.]*))`)
-	matches = regex.FindAllStringSubmatch(string(version), -1)
-	if len(matches) > 0 {
-		ch <- prometheus.MustNewConstMetric(collector.buildInfo, prometheus.GaugeValue, 1, matches[0][3], matches[0][4])
-	}
-
-	strBuilddate := strings.Split(string(version), "/")[2]
-	// Remove newline
-	strBuilddate = strings.Replace(strBuilddate, "\n", "", -1)
+	versionInfo := GetVersionInfo(string(collector.client.Dial(commands.VERSION)))
 
-	// Parse string as date type
-	dateFmt := "Mon Jan 2 15:04:05 2006"
-	builddate, err := time.Parse(dateFmt, strBuilddate)
-
-	if err != nil {
-		log.Error("Error parsing ClamAV date: ", err)
-		ch <- prometheus.MustNewConstMetric(collector.databaseAge, prometheus.GaugeValue, float64(time.Now().Unix()))
-		return
+	if versionInfo.versions_parsed {
```

Comment:
The tricky part with consts is that if we parsed it once but then failed to parse it in the future the old const values will stick around. The only solution would be to delete the existing const so it stops advertising it but that doesn't seem easy. For now we can leave this but it's not ideal.

Reply:
ah hmm...I'm not sure how to go about that - not seeing anything obvious in the docs